Support proxied Terraform registry endpoint (#1057)
hashicorp/go-tfe: CHANGELOG.md

## Enhancements

> * Add support for proxied Terraform Registry endpoints for reading detailed module information by @paladin-devops [#1057](https://github.com/hashicorp/go-tfe/pull/1057) 

Reviewer: ```suggestion
* Adds `ReadTerraformRegistryModule` to support reading a registry module from Terraform Registry's proxied endpoints by @paladin-devops [#1057](https://github.com/hashicorp/go-tfe/pull/1057) 
```

Author: Done!
